### PR TITLE
Add Scaffolding

### PR DESCRIPTION
**Pull request by guseducampos**
This PR attempts to add Scaffolding capabilities to the tool

Missing parts:

- [x] Support more configuration files right now we only are supporting TOML, We need to support JSON and YAML
- [x] Add CLI parameters to accept configuration file

cc @manasv

### CODE REVIEW

File: Sources/MaGusKit/ProjectCommand.swift [hunk 1]
```diff
@@ -54,8 +55,10 @@ public struct ProjectCommand: ParsableCommand {
         let projectGenerator = try ProjectGenerator(outputPath: outputPath,
                                                     projectInformation: information)
 
-        try projectGenerator.generate(for: .all)
-        try projectGenerator.generate(for: .allWith(customs: []))
-        try projectGenerator.generate(for: .custom(templates: []))
+        try projectGenerator.generate(for: .all, loader: DictionaryLoader(templates: Templates.shared.templates))
+        try projectGenerator.generate(for: .allWith(customs: []),
```
Comment (guseducampos) [inline Sources/MaGusKit/ProjectCommand.swift:59]: I need to remove this

File: Package.resolved [hunk 2]
```diff
@@ -36,6 +54,24 @@
           "revision": "35b76bf577d3cc74820f8991894ce3bcdf024ddc",
           "version": "0.0.2"
         }
+      },
+      {
+        "package": "TOMLDecoder",
+        "repositoryURL": "https://github.com/guseducampos/TOMLDecoder.git",
```
Comment (guseducampos) [inline Package.resolved:60]: This will change when decode fix for single value container gets approved by TOMLDecoder author

**Comment by guseducampos** [comment]
@manasv I'll address documentation in a following PR

File: Package.swift [hunk 3]
```diff
@@ -11,8 +11,8 @@ let package = Package(
         .package(url: "https://github.com/stencilproject/Stencil.git", .upToNextMinor(from: "0.13.0")),
         .package(url: "https://github.com/kylef/PathKit", from: "0.9.0"),
         .package(url: "https://github.com/guseducampos/TOMLDecoder.git", .branch("prevent-crash")),
```
Comment (manasv) [inline Package.swift:13]: Are we still using your repo for TOMLDecoder? @guseducampos

Comment (guseducampos) [reply Package.swift:13]: No, I forget change to the main repository.

**Pull request merged**